File: src/main/java/tk/skulk/plugin/core/extensions/entityoverride/EntityOverrideExtension.java
package tk.skulk.plugin.core.extensions.entityoverride;

import org.jetbrains.annotations.NotNull;
import tk.skulk.plugin.core.Plugin;
import tk.skulk.plugin.core.extensions.entityoverride.listeners.CreeperNoGriefListener;
import tk.skulk.plugin.core.extensions.entityoverride.listeners.DragonDropElytraListener;
import tk.skulk.plugin.core.extensions.entityoverride.listeners.EndermanNoGriefListener;
import tk.skulk.plugin.core.extensions.entityoverride.listeners.ExplosionDropAllListener;
import tk.skulk.plugin.wrapper.BaseExtension;

public final class EntityOverrideExtension extends BaseExtension {
    public EntityOverrideExtension(final @NotNull Plugin plugin) {
        super(plugin);
    }

    @Override protected void initListeners() {
        new CreeperNoGriefListener(this);
        new DragonDropElytraListener(this);
        new EndermanNoGriefListener(this);
        new ExplosionDropAllListener(this);
    }
}

File: src/main/kotlin/tk/skulk/plugin/core/extensions/entityOverride/EntityOverrideExtension.kt
package tk.skulk.plugin.core.extensions.entityOverride

import tk.skulk.plugin.core.SNPlugin
import tk.skulk.plugin.core.extensions.entityOverride.listeners.CreeperNoGriefListener
import tk.skulk.plugin.core.extensions.entityOverride.listeners.DragonDropElytraListener
import tk.skulk.plugin.core.extensions.entityOverride.listeners.EndermanNoGriefListener
import tk.skulk.plugin.core.extensions.entityOverride.listeners.ExplosionDropAllListener
import tk.skulk.plugin.framework.Extension

class EntityOverrideExtension(plugin: SNPlugin) : Extension<SNPlugin>(plugin) {
    override val listeners = listOf(
        { CreeperNoGriefListener(this) },
        { DragonDropElytraListener(this) },
        { EndermanNoGriefListener(this) },
        { ExplosionDropAllListener(this) },
    )
}
